File: 프로그래머스/lv2/42842. 카펫/카펫.py
def solution(brown, yellow):
    area = brown + yellow
    for b in range(1, area+1):
        if (area / b) - (area // b) == 0:
            a = area//b
            if b >= a:
                if 2*a + 2*b == brown + 4:
                    return [b,a]

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    const sum = brown + yellow
    
    for(let side = 3; side < sum; side++) {
        const otherSide = sum / side
        const yellowSide = side - 2
        const yellowOtherSide = otherSide - 2
        
        if(yellowSide * yellowOtherSide === yellow) {
            return [Math.max(side, otherSide), Math.min(side, otherSide)]
        }
        
    }
}